keystone_plugin/project.py

```diff
@@ -119,21 +113,32 @@ def validate_users(users, keystone_client):
         keystone_client.users.find(name=user_name)
 
     for user in users:
-        if len(user['roles']) > len(set(user['roles'])):
+        if len(user[ROLES]) > len(set(user[ROLES])):
             msg = 'Roles for user {} are not unique'
             raise NonRecoverableError(msg.format(user['name']))
 
-    role_names = {role for user in users for role in user['roles']}
+    role_names = {role for user in users for role in user[ROLES]}
     for role_name in role_names:
         keystone_client.roles.find(name=role_name)
 
 
+def get_quota(tenant_id, client, what_quota):
+    if what_quota == NEUTRON:
+        quota = client.get_quotas_tenant(tenant=tenant_id)
+        return quota
+    else:
+        quota = client.quotas.get(tenant_id=tenant_id)
+    ctx.logger.info(
+        'Got {0} quota: {1}'.format(what_quota, str(quota)))
+    return quota.to_dict()
+
+
 def update_quota(tenant_id, quota, client, what_quota):
     updated_quota = quota.get(what_quota)
     if updated_quota:
-        if what_quota == 'neutron':
+        if what_quota == NEUTRON:
             new_quota = client.update_quota(tenant_id=tenant_id,
-                                            body={'quota': updated_quota})
+                                            body={QUOTA: updated_quota})
         else:
             new_quota = client.quotas.update(tenant_id=tenant_id,
                                              **updated_quota)
```

Comment:
I would like to see some exception handling, either logging or raising what happened here.